spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseDeleteOrphanFilesSparkAction.java

```diff
@@ -232,6 +259,75 @@ private DeleteOrphanFiles.Result doExecute() {
+    if (equalSchemes != null && !equalSchemes.isEmpty()) {
+      String firstScheme = equalSchemes.get(0);
+      equalSchemes.forEach(value -> equalSchemeAndAuthoritiesMap.put(value, firstScheme));
+    }
+
+    if (equalAuthorities != null && !equalAuthorities.isEmpty()) {
+      String firstAuthority = equalAuthorities.get(0);
+      equalAuthorities.forEach(value -> equalSchemeAndAuthoritiesMap.put(value, firstAuthority));
+    }
+    return equalSchemeAndAuthoritiesMap;
+  }
+
+  @VisibleForTesting
+  static List<String> findOrphanFiles(Dataset<Row> actualFileDF,
+                                      Dataset<Row> validFileDF,
+                                      Map<String, String> equalSchemes,
+                                      Map<String, String> equalAuthorities,
+                                      SetAccumulator<Pair<String, String>> setAccumulator,
+                                      PrefixMismatchMode prefixMismatchMode) {
+    Map<String, String> equalSchemesAndAuthoritiesMap = Maps.newHashMap();
+    equalSchemesAndAuthoritiesMap.putAll(flattenMap(equalSchemes));
+    equalSchemesAndAuthoritiesMap.putAll(flattenMap(equalAuthorities));
+
+    Dataset<Row> normalizedActualFileDF = actualFileDF.mapPartitions(toFileMetadata(equalSchemesAndAuthoritiesMap),
+        RowEncoder.apply(FILE_METADATA_SCHEMA)).as("actual");
+    Dataset<Row> normalizedValidFileDF = validFileDF.mapPartitions(toFileMetadata(equalSchemesAndAuthoritiesMap),
+        RowEncoder.apply(FILE_METADATA_SCHEMA)).as("valid");
+
+    Column actualFileName = normalizedActualFileDF.col("path");
+    Column validFileName = normalizedValidFileDF.col("path");
+
+    Dataset<FileDescriptor> mayBeOrphanFilesDF = normalizedActualFileDF.join(normalizedValidFileDF,
+        actualFileName.equalTo(validFileName),
+        "leftouter")
+        .selectExpr("actual.scheme as actualScheme",
+            "actual.authority as actualAuthority",
+            "actual.path as actualPath",
+            "actual.file_path as actualFilePath",
+            "valid.scheme as validScheme",
+            "valid.authority as validAuthority",
+            "valid.path as validPath",
+            "valid.file_path as validFilePath")
+        .as(Encoders.bean(FileDescriptor.class));
+
+    List<String> orphanFiles = mayBeOrphanFilesDF.mapPartitions(
+        new MapOrphanFilesFunction(setAccumulator), Encoders.STRING()).collectAsList();
+    if (prefixMismatchMode == PrefixMismatchMode.ERROR && !setAccumulator.value().isEmpty()) {
+      throw new ValidationException("Unable to deterministically find all orphan files." +
+          " Found file paths that have same file path but different authorities/schemes. Conflicting" +
+          " authorities/schemes found: %s", setAccumulator.value().toString());
+    }
+    return orphanFiles;
+  }
+
+  private static Map<String, String> flattenMap(Map<String, String> toBeFlattenedMap) {
```

Comment:
To be honest, I am not sure supporting comma separated values will be easier for the user. I'd either skip this logic and just support a plain map or at least support commas in the key vs in the value (`s3a, s3n` -> `s3`).